functions/scripts/generate-packing-slips.js

```diff
@@ -0,0 +1,192 @@
+const fs = require('fs');
+const { Readable, finished } = require('stream');
+
+const _ = require('lodash');
+const markdownpdf = require('markdown-pdf');
+const pdfmerge = require('easy-pdf-merge');
+const util = require('util');
+
+const {
+  BULK_ORDER_TABLE,
+  INTAKE_TABLE,
+  ITEMS_BY_HOUSEHOLD_SIZE_TABLE,
+  getAllRecords,
+  getRecordsWithStatus,
+  getBulkOrder,
+} = require('../airtable');
+
+class PackingSlip {
+  constructor(intakeRecord, requested, provided) {
+    this.intakeRecord = intakeRecord;
+    this.requested = requested;
+    this.provided = provided;
+  }
+
+  getMarkdown(itemToCategory) {
+    const fields = this.intakeRecord[1];
+
+    let markdown = `# **${fields.ticketID}**: ${fields.requestName} (${fields.nearestIntersection})\n\n`;
+
+    _.forEach(
+      _.toPairs(
+        _.groupBy(
+          _.toPairs(this.provided),
+          ([item,]) => { return itemToCategory[item]; },
+        )
+      ),
+      ([category, provided]) => {
+        markdown += `### **${category}**:\n`;
+        _.forEach(
+          provided,
+          ([item, numProvided]) => {
+            markdown += `\t- ${numProvided} ${item}\n`;
+          }
+        );
+        markdown += '\n';
+      },
+    );
+
+    if (!_.isNull(fields.otherItems) || !_.isEqual(this.provided, this.requested)) {
+      markdown += '\n---\n';
+      markdown += '## **Other** (Provided By Bed-Stuy Strong!):\n';
+
+      if (!_.isEqual(this.provided, this.requested)) {
+        markdown += '### Missing:\n';
+
+        const missing = _.filter(
+          _.map(
+            _.toPairs(this.requested),
+            ([item, numRequested]) => {
+              return [item, numRequested - _.get(this.provided, item, 0)];
+            },
+          ),
+          ([, diff]) => { return diff !== 0; },
+        );
+
+        _.forEach(
+          missing,
+          ([item, numMissing]) => {
+            markdown += `\t- ${numMissing} ${item}\n`;
+          }
+        );
+      }
+
+      if (!_.isNull(fields.otherItems)) {
+        markdown +=  `### Custom Items: ${fields.otherItems}\n`;
+      }
+    }
+
+    return markdown;
+  }
+}
+
+async function savePackingSlips(packingSlips) {
+  const itemToCategory = _.fromPairs(
+    _.map(
+      await getAllRecords(ITEMS_BY_HOUSEHOLD_SIZE_TABLE),
+      ([, fields,]) => { return [fields.item, fields.category]; },
+    )
+  );
+
+  // TODO : remove the out directory
+
+  await fs.promises.mkdir('out/');
```

Comment:
are you running this locally with a different version of node than the emulators? i didn't think `fs.promises` was available in node 8

Reply:
Yes, I'm running with a newer version of node locally. I'll leave it for now and hope that we get to a newer version of node soon 😝